File: exhibits/new-exhibit.mjs
#!/usr/bin/env node
/** Interactive CLI: create exhibit from _template. */
import fs from 'node:fs'
import path from 'node:path'
import readline from 'node:readline'
import { spawn } from 'node:child_process'
import { fileURLToPath } from 'node:url'
import { createExhibit, suggestNextExhibitDir } from './exhibit-create.mjs'

const ROOT = path.resolve(path.dirname(fileURLToPath(import.meta.url)))
const args = process.argv.slice(2)

function ask(q, def = '') {
  const rl = readline.createInterface({ input: process.stdin, output: process.stdout })
  const prompt = def ? `${q} [${def}]: ` : `${q}: `
  return new Promise(resolve => rl.question(prompt, ans => { rl.close(); resolve((ans || def).trim()) }))
}

async function main() {
  let dir = args[0]
  let title = args[1]
  let subtitle = args[2] || ''

  const suggested = suggestNextExhibitDir(ROOT)
  if (!dir) dir = await ask('展品编号（如 005 或 craft-005）', suggested.replace(/^craft-/i, ''))
  if (!title) title = await ask('展品名称（左上角显示）')

  const result = createExhibit(ROOT, { dir, title, subtitle })
  console.log('')
  console.log('[OK] 已创建', result.dir)
  console.log('  名称:', result.title)
  console.log('  目录:', path.join(ROOT, result.dir))
  console.log('')
  console.log('下一步：')
  console.log('  1. 把 model.glb 放入', path.join(result.dir, 'assets'))
  console.log('  2. 刷新工作台 → 点「编辑」→ 调相机/热点 → 保存')
  console.log('')
  if (process.platform === 'win32') {
    spawn('explorer.exe', [result.assetsDir], { detached: true, stdio: 'ignore' }).unref()
  }
}

main().catch(e => {
  console.error('[ERROR]', e.message)
  process.exit(1)
})

File: exhibits/_launch/new-exhibit.py
#!/usr/bin/env python3
"""Create exhibit from _template (Python CLI for portable installs)."""
from __future__ import annotations

import os
import subprocess
import sys
from pathlib import Path

ROOT = Path(__file__).resolve().parent.parent
sys.path.insert(0, str(ROOT))
from exhibit_create import create_exhibit, suggest_next_exhibit_dir  # noqa: E402


def main() -> int:
    args = sys.argv[1:]
    try:
        suggested = suggest_next_exhibit_dir(ROOT)
        default_num = suggested.replace('craft-', '', 1)
    except ValueError:
        default_num = ''
    dir_name = args[0] if args else input(
        f'展品编号（如 005 或 craft-005） [{default_num}]: '
    ).strip() or default_num
    title = args[1] if len(args) > 1 else input('展品名称（左上角显示）: ').strip()
    subtitle = args[2] if len(args) > 2 else ''
    try:
        result = create_exhibit(ROOT, dir_name, title, subtitle)
    except Exception as e:
        print(f'[ERROR] {e}', file=sys.stderr)
        return 1
    print('')
    print('[OK] 已创建', result['dir'])
    print('  名称:', result['title'])
    print('  目录:', ROOT / result['dir'])
    print('')
    print('下一步：')
    print('  1. 把 model.glb、panorama.jpg 放入', result['dir'] + '/assets')
    print('  2. 刷新工作台 → 点「编辑」→ 调相机/热点 → 保存')
    print('')
    if os.name == 'nt':
        subprocess.Popen(['explorer', str(ROOT / result['assetsDir'])])
    return 0


if __name__ == '__main__':
    raise SystemExit(main())
